src/main/java/com/rarchives/ripme/ripper/rippers/RedditRipper.java

```diff
@@ -149,9 +155,28 @@ private JSONArray getJsonArrayFromURL(URL url) throws IOException {
         return jsonArray;
     }
 
+    /**
+     * Turns child JSONObject's into usable URLs and hands them off for further processing
+     * Performs filtering checks based on the reddit.
+     * Only called from getAndParseAndReturnNext() while parsing the JSONArray returned from reddit's API
+     * @param child The child to process
+     */
     private void parseJsonChild(JSONObject child) {
         String kind = child.getString("kind");
         JSONObject data = child.getJSONObject("data");
+
+        //Upvote filtering
+        if (Utils.getConfigBoolean("reddit.rip_by_upvote", false)){
+            int score = data.getInt("score");
+            int maxScore = Utils.getConfigInteger("reddit.max_upvotes", Integer.MAX_VALUE);
+            int minScore = Utils.getConfigInteger("reddit.min_upvotes", Integer.MIN_VALUE);
+
+            if (score > maxScore || score < minScore) {
+
+                return; //Outside specified range, do not download
```

Comment:
You should print a debug message here. Otherwise users might forget they have this enabled and then wonder why certain files aren't being downloaded

Reply:
```suggestion
                String message = "Skipping post with score outside specified range of " + minScore + " to " + maxScore;
                LOGGER.debug(message);
                return; //Outside specified range, do not download
```
Like so?

Reply:
Looks good to me